### PR TITLE
unify the common methods for DistributedStorage

### PR DESCRIPTION
**Pull request by yito88**
Before adding a prefix to each namespace, I'd like to unify the common methods for `DistributedStorage`.

- `setTargetToIfNot()` and `checkIfPrimaryKeyExists()` in each storage are unified in `StorageUtility`
- Add `TableMetadata` interface and rename each `TableMetadata`

### CODE REVIEW

**Review by feeblefakie — COMMENTED** [review]
Overall looking good! 
Left one fix and one suggestion.

File: src/main/java/com/scalar/db/storage/StorageUtility.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+import com.scalar.db.api.Mutation;
+import com.scalar.db.api.Operation;
+import com.scalar.db.io.Key;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Utilities for {@link DistributedStorage}
```
[the remark this answers is not in the document]
Comment (yito88) [reply src/main/java/com/scalar/db/storage/StorageUtility.java]: Fixed!

File: src/main/java/com/scalar/db/storage/StorageUtility.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ * @author Yuji Ito
+ */
+public final class StorageUtility {
```
[the remark this answers is not in the document]
Comment (yito88) [reply src/main/java/com/scalar/db/storage/StorageUtility.java]: Thank you for the suggestion! Looks better. Modified.

**Review by feeblefakie — APPROVED** [review]
LGTM!

**Pull request merged**